auth/src/api/handlers.rs

```diff
@@ -91,6 +94,45 @@ pub(crate) async fn add_subscription_items(
     Ok(())
 }
 
+#[instrument(skip(claim, user_manager), fields(account.name = claim.sub, account.tier = %claim.tier, account.subscription_id = tracing::field::Empty))]
+pub(crate) async fn delete_subscription_items(
+    Extension(claim): Extension<Claim>,
+    State(user_manager): State<UserManagerState>,
+    Path(metadata_id): Path<String>,
+) -> Result<Json<SubscriptionResponse>, Error> {
+    let user = user_manager.get_user(AccountName::from(claim.sub)).await?;
+
+    let Some(ref subscription_id) = user.subscription_id else {
+        return Ok(Json(SubscriptionResponse {
+            success: false,
+            message: "subscription does not exist".to_string(),
+        }));
+    };
+
+    span::Span::current().record("account.subscription_id", subscription_id.as_str());
+
+    match user_manager
+        .delete_subscription_item(subscription_id, &metadata_id)
+        .await
+    {
+        Ok(_) => Ok(Json(SubscriptionResponse {
+            success: true,
+            message: "successfully deleted subscription item".to_string(),
+        })),
+        // If the subscription item is missing, we still return 200, since the rds instance could
+        // have been provisioned before we implemented billing. If the subscription is canceled,
```

Comment:
Can you explain that part about RDS a bit more ? I don't understand how this can happen.

Reply:
So in the past anyone could provision RDS, and currently only Pro tier users can provision it. That means we have quite a few users with an RDS instance, but without a subscription. We don't want to block them from deleting their RDS instance because of that.

There are also Pro tier users who have a subscription, but when they provisioned RDS it was still unpaid, no subscription item was added. We will be adding an item to their subscriptions manually following this release and an announcement with a warning, but there will be some time where we will have users with subscriptions and RDS but no RDS subscription item. I can expand the comment with a note that we should not return 200 for missing sub item after that point, since it would then be a bug.

Reply:
Great explanation, all clear now, thanks !